src/github.com/couchbase/sync_gateway/db/channel_cache.go

```diff
@@ -257,3 +307,140 @@ func (c *channelCache) prependChanges(changes LogEntries, changesValidFrom uint6
 		return 0
 	}
 }
+
+type lateLogEntry struct {
+	logEntry      *LogEntry
+	arrived       time.Time
+	listenerCount uint64
```

Comment:
can you add an explanation of why the `listenerCount` field is needed?  (what overall problem does it solve)